### ISSUE TITLE
www.canva.com - viewing a design gives a black page

### ISSUE THREAD

**Issue by bitnimble** [AdguardTeam/AdguardFilters#147371]
### Prerequisites

- [X] This site DOES NOT contain sexually explicit material, otherwise use [NSFW-specific form](./new?template=bug_report_NSFW.yml);
- [X] Filters were updated before reproducing an issue;
- [X] AdGuard product version is up-to-date;
- [X] Browser version is up-to-date;
- [X] If the site or app is broken, disabling AdGuard protection resolves an issue.

### What product do you use?

Other ad blocker

### AdGuard version

Annoyances 2.1.54.53

### What type of problem have you encountered?

Website or app doesn't work properly

### Which browser(s) do you use?

Chrome

### Which device do you use?

Desktop

### Where is the problem encountered?

https://www.canva.com/design/DAFfIKR3n7I/hm3-tYtgAaNqcWOvPrf45Q/view

### What filters do you have enabled?

AdGuard Annoyances filter

### Add your comment and screenshots

1. Caused by a mobile-targeting `canva.com#$#body { overflow: auto !important; }` rule, which hides the entire body

2. Screenshots

Screenshot is a little bit useless in this case, as the entire page is black 😅

### Privacy

- [X] I agree to follow this condition

**Comment by AdamWr** [comment]
I suppose that it's related to this issue - https://github.com/AdguardTeam/AdguardFilters/issues/146275#issuecomment-1492863178 and it's caused by a different rule.
Could you please add this rule:
```adblock
canva.com#@#.NqU4Yg
```
to user rules and then check if issue still occurs?

**Comment by krystian3w** [comment]
Error 404, Account required?

**Comment by bitnimble** [comment]
>Error 404, Account required?

Argh, derp, that's just a brain fart. I put a placeholder link for myself to update later, then went to lunch and promptly forgot about it when I came back to hit the Submit button. I've updated the link now! 👍 Here is a video instead demonstrating the issue:

https://user-images.githubusercontent.com/4076797/229735020-96ed649f-b091-4aaf-af51-841d56f301fd.mp4

>I suppose that it's related to this issue

Ahh yes, I didn't see that - can confirm that that exception works! `body` was a red herring, my apologies 😄  Tested both and confirmed that `.NqU4Yg` is the problematic rule. Thanks for fixing!

Disclosure/source: I work at Canva. I didn't look deeply yet but I think that selector targets the root node for our modal / flyout layer, so it affects all kinds of dropdowns etc. I see that you've updated it now to be more specific 👍 

Also as a side note for future cases -- at Canva we use a fairly aggressive CSS classname minifier, which is not guaranteed to be stable across builds. That means that potentially even up to once a day, classnames could change if that area of the code is modified. Apart from the fact that it's breaking other flyouts, I wouldn't trust a selector like `.NqU4Yg` to actually work as intended for more than a few weeks. If it's not too deep in the DOM, a structural / path selector instead of a specific classname would be more reliable 👍 

Will close this one now, cheers!
